### PR TITLE
feat: Update addons to latest supported versions

### LINKED ISSUE #1093: Update addon versions to be current
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

- Update addons to use the current (latest) version

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?

- Update addons to latest supported versions
- Update `tflint` to current version and correct linting warnings (mostly around use of `*`)
- Remove unnecessary provider requirements

### Motivation

- Resolves #1093
- Closes #1091
- Closes #1092

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by Zvikan — APPROVED** [review]
LGTM, minor comments and question, also, let's make sure to run some e2e for sanity check.
Thank you!

File: modules/kubernetes-addons/argocd/locals.tf [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
+  # https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml
   default_helm_config = {
     name             = local.name
     chart            = local.name
     repository       = "https://argoproj.github.io/argo-helm"
-    version          = "4.9.14"
+    version          = "5.8.3"
     namespace        = local.namespace
-    timeout          = 1200
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply modules/kubernetes-addons/argocd/locals.tf:13]: the helm-addon already defaults to this so its redundant here https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/0529549fa2f4b07f65c199c2573b81b2e979aa56/modules/kubernetes-addons/helm-addon/main.tf#L7

File: modules/kubernetes-addons/aws-load-balancer-controller/locals.tf [hunk 2]
```diff
@@ -2,13 +2,13 @@ locals {
...
 
+  # https://github.com/aws/eks-charts/blob/master/stable/aws-load-balancer-controller/Chart.yaml
   default_helm_config = {
     name        = local.name
     chart       = local.name
     repository  = "https://aws.github.io/eks-charts"
-    version     = "1.4.3"
+    version     = "1.4.5"
     namespace   = "kube-system"
-    timeout     = "1200"
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply modules/kubernetes-addons/aws-load-balancer-controller/locals.tf:11]: default already for helm-addon https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/0529549fa2f4b07f65c199c2573b81b2e979aa56/modules/kubernetes-addons/helm-addon/main.tf#L7

File: modules/kubernetes-addons/ingress-nginx/outputs.tf [hunk 3]
```diff
@@ -1,4 +1,4 @@
 output "argocd_gitops_config" {
   description = "Configuration used for managing the add-on with ArgoCD"
-  value       = var.manage_via_gitops ? local.argocd_gitops_config : null
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply modules/kubernetes-addons/ingress-nginx/outputs.tf:3]: Could you elaborate a bit more? This is all that was behind the local variable https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1096/files/b838d60698d590d25936fa3747b6817ccb9cd5d7#diff-646886bed0c86a54e6455e85bc6760a1674984bbe88fea71bc7758044df75844L24-L26

[the remark this answers is not in the document]
Comment (Zvikan) [reply modules/kubernetes-addons/ingress-nginx/outputs.tf:3]: You are right, I read this wrong, I expected to see more dynamic configuration but it's local that mostly have enable=true and in some cases the service account name.

**Comment by bryantbiggs** [comment]
all tests pass save for the usual suspect failures https://github.com/aws-ia/terraform-aws-eks-blueprints/actions/runs/3348792340/jobs/5548211087

**Pull request merged**